File: src/main/java/com/atsuishio/superbwarfare/command/CommandRegister.java
package com.atsuishio.superbwarfare.command;

import net.fabricmc.fabric.api.command.v2.CommandRegistrationCallback;
import net.minecraft.commands.Commands;

public class CommandRegister {
    public static void registerEvents() {
        CommandRegistrationCallback.EVENT.register((dispatcher, registryAccess, environment) -> registerCommand(dispatcher));
    }

    private static void registerCommand(com.mojang.brigadier.CommandDispatcher<net.minecraft.commands.CommandSourceStack> dispatcher) {
        var command = Commands.literal("sbw");
        command.then(AmmoCommand.get());
        command.then(ConfigCommand.get());
        command.then(TDMCommand.get());

        var result = dispatcher.register(command);
        dispatcher.register(Commands.literal("superbwarfare").redirect(result));
    }
}

File: src/main/kotlin/com/atsuishio/superbwarfare/command/CommandRegister.kt
package com.atsuishio.superbwarfare.command

import com.atsuishio.superbwarfare.command.builder.buildCommand
import com.mojang.brigadier.builder.LiteralArgumentBuilder
import net.fabricmc.fabric.api.command.v2.CommandRegistrationCallback
import net.minecraft.commands.CommandSourceStack
import net.minecraft.commands.Commands

object CommandRegister {
    @JvmStatic
    fun register() {
        CommandRegistrationCallback.EVENT.register { dispatcher, registryAccess, environment ->
            val command = buildCommand("sbw") {
                add(AMMO_COMMAND)
                add(CONFIG_COMMAND)
                add(TDM_COMMAND)
                add(RIDE_COMMAND)
                add(DISMOUNT_COMMAND)
                add(SKIN_COMMAND)
                add(LOITER_COMMAND)
            }

            val result = dispatcher.register(command as LiteralArgumentBuilder<CommandSourceStack>)
            dispatcher.register(Commands.literal("superbwarfare").redirect(result))
        }
    }
}